source/plugin/data/custom/customdata.rst

```diff
@@ -0,0 +1,296 @@
+for a single key. The following example does the same as the above example.
+
+.. code-block:: java
+
+    DataStore datastore = DataStore.of(key, DataQuery.of(), Entity.class);
+
+Multi-Key Data Store
+====================
+
+If you are registering multiple keys onto a single ``DataStore`` then the first approach should be used, however the
+other keys should be specified with the original key, such as the following example.
+
+.. code-block:: java
+
+    import org.spongepowered.api.entity.Entity;
+
+    DataStore datastore = DataStore
+        .builder()
+        .pluginData(resourceKey)
+        .holder(Entity.class)
+        .key(key)
+        .key(innerKey, DataQuery.of("inner_data"))
+        .build();
+
+Data Provider
+=============
+
+For data that requires more code to be used whenever the getter, setter, deleter are used will require the use of
+a ``DataProvider``. With a ``dataProvider`` a plugin is able to manipulate how its data should be received, set and
+deleted automaticly. 
+
+In the following example, we will be getting the UUID from the last attacker but if there is no last attacker, then
+return the player's UUID instead.
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataProvider;
+
+    DataProvider<Value<UUID>, UUID> provider = DataProvider.mutableBuilder()
+        .dataKey(innerKey)
+        .dataHolder(ServerPlayer.class)
+        .get(this::myCustomGetter)
+        .build();
+
+    public UUID myCustomGetter(ServerPlayer player){
+        return player.get(key).orElse(player.uniqueId());
+    }
+
+.. note::
+
+    Data Provider's are completely optional, if your data does not require one then don't use one
+
+.. tip::
+
+    Data Providers are great if you wish to have your data be synced with a database
+
+
+Data Registration
+=================
+
+The final object you will need to register your data is the :javadoc:`DataRegistration` which combines 
+your ``Key``, ``DataStore`` and ``DataProvider`` together into a single package that you can register.
+
+.. code-block:: java
+
+    import org.spongepowered.api.data.DataRegistration;
+
+    DataRegistration myData = DataRegistration.builder()
+        .key(key)
+        .store(datastore)
+        .provider(provider)
+        .build();
+
+    event.register(myData);
+
+Data Builder Register
+=====================
+
+The final part of your custom data registration is registering the data builder so your data can be
+constructed upon reboot. This is registered though the :javadoc:`DataManager`, although it is recommended
+that you register it within the ``RegisterDataEvent``.
+
+.. code-block:: java
+
+    Sponge.dataManager().registerBuilder(LastAttackerDataSerilizable.class, new LastAttackerDataBuilder());
+
+Simple Custom Data
+==================
+
+All of above is a lot of work if you are just wanting to register a java primitive or ``String`` to
```

Comment:
"if you are just wanting to" -> "if you just want to"

Reply:
Updated